File: algorithm/Two_Sum/Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map={}
        complement = 0
        for i in range(len(nums)):
            complement = target-nums[i]
            if complement in hash_map :
                return [hash_map[complement],i]
            hash_map[nums[i]]=i
        return []

File: algorithm/Two_Sum/Two_Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> hash_map = new HashMap<Integer,Integer>();
        int complement;
        for(int i=0;i<nums.length;i++){
            complement = target - nums[i];
            if(hash_map.containsKey(complement)){
                return new int[]{hash_map.get(complement),i};
            }
            hash_map.put(nums[i],i);
        }
        return null;
    }
}